contracts/test/unit/oracle/VoltSystemOracle.t.sol

```diff
@@ -0,0 +1,212 @@
+            startTime,
+            startPrice
+        );
+    }
+
+    function testSetup() public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        assertEq(
+            voltSystemOracle.annualChangeRateBasisPoints(),
+            annualChangeRateBasisPoints
+        );
+        assertEq(voltSystemOracle.oracleStartTime(), startTime);
+        assertEq(voltSystemOracle.startTime(), block.timestamp);
+        assertEq(voltSystemOracle.getCurrentOraclePrice(), startPrice);
+    }
+
+    function testInitFailsBeforeStartTime() public {
+        vm.expectRevert("VoltSystemOracle: not past start time");
+
+        voltSystemOracle.init();
+    }
+
+    function testSecondInitFails() public {
+        vm.warp(voltSystemOracle.oracleStartTime());
+
+        voltSystemOracle.init();
+
+        vm.expectRevert("Initializable: contract is already initialized");
+        voltSystemOracle.init();
+    }
+
+    function testCompoundBeforeInitFails() public {
+        vm.expectRevert("Timed: time not ended");
+        voltSystemOracle.compoundInterest();
+    }
+
+    function testInitSucceeds() public {
+        assertEq(voltSystemOracle.startTime(), 1); /// start time starts at block.timestamp
+        vm.warp(voltSystemOracle.oracleStartTime());
+
+        voltSystemOracle.init();
+
+        /// start time is successfully set
+        assertEq(
+            voltSystemOracle.startTime(),
+            voltSystemOracle.oracleStartTime()
+        );
+    }
+
+    function testCompoundSucceedsAfterOneYear() public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        vm.warp(block.timestamp + voltSystemOracle.oracleStartTime());
+
+        voltSystemOracle.init();
+        uint256 oraclePrice = voltSystemOracle.oraclePrice();
+
+        vm.warp(block.timestamp + 365 days);
+
+        assertEq(
+            voltSystemOracle.getCurrentOraclePrice(),
+            (oraclePrice *
+                (Constants.BASIS_POINTS_GRANULARITY +
+                    annualChangeRateBasisPoints)) /
+                Constants.BASIS_POINTS_GRANULARITY
+        );
+        voltSystemOracle.compoundInterest();
+        assertEq(
+            voltSystemOracle.oraclePrice(),
+            (oraclePrice *
+                (Constants.BASIS_POINTS_GRANULARITY +
+                    annualChangeRateBasisPoints)) /
+                Constants.BASIS_POINTS_GRANULARITY
+        );
+    }
+
+    function testLinearInterpolation() public {
+        assertEq(voltSystemOracle.oraclePrice(), startPrice);
+        vm.warp(voltSystemOracle.oracleStartTime() + 365 days);
+
+        assertEq(
+            voltSystemOracle.getCurrentOraclePrice(),
+            (startPrice *
+                (Constants.BASIS_POINTS_GRANULARITY +
+                    annualChangeRateBasisPoints)) /
+                Constants.BASIS_POINTS_GRANULARITY
+        );
+    }
+
+    function testLinearInterpolationFuzz(uint32 timeIncrease) public {
```

Comment:
Does this have clamps on the times? If not the >= 365 days seems likely to be chosen most of the time which isn't interesting

Reply:
I could create another test that does have bounds on the time to specifically test the under a year logic. I do believe that the current fuzzing setup tests both the lower and upper bounds of the parameters, so it will always pass numbers close to the min and close to the max.

Reply:
added